### PR TITLE
Avoid setting purge_seq in local doc with false

### PR DESCRIPTION
**Pull request by jiangphcn**
## Overview

During initial phase, the default value for purge_seq of local purge document for geo spatial index should be purge sequence of database if it doesn't exist before, and should not be false. This PR is aimed to avoid such setting to false.

## Testing recommendations

make check skip_deps+=couch_epi apps=hastings tests=hastings_purge_test_

```
======================== EUnit ========================
Hastings Purge Tests
Application crypto was left running!
  hastings_purge_docs_test:69: test_purge_single...[0.260 s] ok
  hastings_purge_docs_test:84: test_purge_multiple...[0.197 s] ok
  hastings_purge_docs_test:103: test_purge_multiple2...[0.234 s] ok
  hastings_purge_docs_test:129: test_purge_local_purge_doc...[0.276 s] ok
  hastings_purge_docs_test:210: test_purge_verify_index...[0.160 s] ok
  hastings_purge_docs_test:240: test_purge_remove_local_purge_doc...[0.189 s] ok
  hastings_purge_docs_test:280: test_purge_hook_before_compaction...[0.832 s] ok
erl_child_setup: failed with error 32 on line 240
[os_mon] cpu supervisor port (cpu_sup): Erlang has closed
  [done in 3.604 s]
=======================================================
  All 7 tests passed.
```

## Related Issues or Pull Requests

COUCHDB-3326

## Checklist

- [X] Code is written and works correctly;
- [X] Changes are covered by tests;
- [X] Documentation reflects the changes;

### CODE REVIEW

File: src/hastings_util.erl [hunk 1]
```diff
@@ -147,6 +147,7 @@ maybe_create_local_purge_doc(Db, Index) ->
         {not_found, _Reason} ->
             DefaultPurgeSeq = couch_db:get_purge_seq(Db),
             PurgeSeq = try easton_index:get(Index#h_idx.pid, purge_seq) of
+                false -> DefaultPurgeSeq;
```
Comment (tonysun83) [inline src/hastings_util.erl]: can it be possible that the index `purge_seq` exists, but something happened with easton and it returns `false`, and we return DB purge_seq which is different than index purge_seq?

Comment (davisp) [reply src/hastings_util.erl]: Its always possible that Easton could like about its purge_seq but that'd be a bug in Easton and not this code relying on it, I think?

Comment (tonysun83) [reply src/hastings_util.erl]: @davisp @jiangphcn : so iiuc, `false` in this case means that there is no existing value for `purge_seq` for this particular index. Unfortunately, I don't understand purge well enough to know how `PurgeSeq` will be used later on. So what is the importance of assigning it to default db purge_seq value vs say 0?

**Comment by tonysun83** [comment]
+1

File: src/hastings_util.erl (hunk 1 above)
Comment (tonysun83) [reply src/hastings_util.erl]: understood, thanks for clarifying

**Pull request merged**